jacodb-taint-configuration/src/main/kotlin/org/jacodb/taint/configuration/Position.kt

```diff
@@ -23,31 +23,55 @@ fun interface PositionResolver<out R> {
     fun resolve(position: Position): R
 }
 
+/**
+ * A class representing a position of tainted data in a method.
+ */
 @Serializable
 sealed interface Position
 
+/**
+ * Represents an argument of a method call.
+ * Numeration starts from zero, `this` parameter is not included. <p>
+ *
+ * For instance, `obj.foo(a, b)` -> a := Argument(0), b := Argument(1)
+ */
 @Serializable
 @SerialName("Argument")
 data class Argument(@SerialName("number") val index: Int) : Position
 
+/**
+ * Represents any argument of a method call except `this` instance,
+ * This is a short form for a set of [Argument]s with all indices of the method parameters.
+ */
 @Serializable
 @SerialName("AnyArgument")
 object AnyArgument : Position {
     override fun toString(): String = javaClass.simpleName
 }
 
+/**
+ * Represents `this` argument of a method call.
+ */
 @Serializable
 @SerialName("This")
 object This : Position {
     override fun toString(): String = javaClass.simpleName
 }
 
+/**
+ * Represents a resulting value of a method call.
+ * It is for regularly returned objects only, and it is not suitable for thrown exceptions.
+ */
 @Serializable
 @SerialName("Result")
 object Result : Position {
     override fun toString(): String = javaClass.simpleName
 }
 
+/**
+ * Represents any element of the collection (string, array, list, etc.),
```

Comment:
This description is "technically correct", but is not very helpful to mirror the reality and current implementation. We only interpret `ResultAnyElement` position as "any element of an array". Currently, it is not possible to express "all elements of a collection (e.g. List) are tainted, but the collection itself is not" using this Position alone — for collections, we simply over-approximate it via `PassThrough` rules, e.g. on `List::get`: `CopyAllMarks(This, Result)`, without relying on `ResultAnyElement`.

Reply:
The configuration doesn't depend on the current implementation since there are at least two different implementations: here and in USVM. So, the question is "What is this supposed to mean in the perfect world?". As far as I understood, this is exactly what is written -- position somewhere in the array: any element of the array might be the one.